File: ruby/find_words.rb
# frozen_string_literal: true

# @param {String[]} words
# @return {String[]}
def find_words(words)
  rows = %w[qwertyuiop asdfghjkl zxcvbnm]
  result = []

  words.each do |word|
    next unless word.downcase.chars.all? { |c| rows[0].include?(c) } ||
                word.downcase.chars.all? { |c| rows[1].include?(c) } ||
                word.downcase.chars.all? { |c| rows[2].include?(c) }

    result << word
  end

  result
end

File: typescript/findWords.ts
function findWords(words: string[]): string[] {
  const rows: string[] = ["qwertyuiop", "asdfghjkl", "zxcvbnm"];
  const result: string[] = [];

  words.forEach((word) => {
    const lowercaseWord = word.toLowerCase();
    const isInRow =
      lowercaseWord.split("").every((c) => rows[0].includes(c)) ||
      lowercaseWord.split("").every((c) => rows[1].includes(c)) ||
      lowercaseWord.split("").every((c) => rows[2].includes(c));

    if (isInRow) {
      result.push(word);
    }
  });

  return result;
}
